docs/dev/webapp/components.md

```diff
@@ -0,0 +1,98 @@
+# Components
+**Note:** This guide assumes you are in the `app/src/app/` folder.
+
+In Angular components are used to seperate an application in different display blocks that display data, listen on user input, and take action based on that input.
+
+In the following all components of the TOSCAna web app will be shortly explained:
+
+## Overview
+
+![](img/components.png)
+## app
+
+![](img/app-comonent.png)
+
+The app component is the entry point of the web app it holds the topbar, sidebar and the view.
+
+**View:** The content of the view changes dynamically based on the route. The `routes` array in `modules/routing/routing.modules.ts` maps a route to a component.
+You can read more about routing [here](https://angular.io/guide/router) in the Angular docs.
+
+## csar-sidebar 
+
+![](img/sidebar.png)
+
+The sidebar shows all available CSARs if you click on a CSAR it also shows its sub items.
+
+The sidebar component also manages CSAR upload it checks if the CSAR name is valid and if the User submits it uploads the file to the server.
```

Comment:
`CSAR upload it checks` -> `CSAR upload. It checks`

Reply:
`if the User submits it uploads the file to the server` -> `when the user submits, the file gets uploaded to the server.`